soh/soh/OTRGlobals.cpp

```diff
@@ -1718,6 +1746,46 @@ Color_RGB8 GetColorForControllerLED() {
                     break;
             }
         }
+        if (gPlayState && (source == LED_SOURCE_NAVI_ORIGINAL || source == LED_SOURCE_NAVI_COSMETICS)) {
+            Actor* arrowPointedActor = gPlayState->actorCtx.targetCtx.arrowPointedActor;
+            if (arrowPointedActor) {
+                ActorCategory category = (ActorCategory)arrowPointedActor->category;
+                switch (category) {
+                    case ACTORCAT_PLAYER:
+                        if (source == LED_SOURCE_NAVI_COSMETICS &&
+                            CVarGetInteger("gCosmetics.Navi_IdlePrimary.Changed", 0)) {
+                            color = CVarGetColor24("gCosmetics.Navi_IdlePrimary.Value", defaultIdleColor.inner);
+                            break;
```

Comment:
Actually I think this should be outside the if statement

Reply:
The break that is

Reply:
Not sure how to best approach it, but since I believe the NPC cosmetic may not have changed but the enemy one may be changed, then this can lead to the NPC falling into the boss case and using that color. The simple fix is to just add the `color = LEDColorDefaultNaviColorList[category].inner;` at the end of every case followed by a `break;` or putting it in an else statement with the break outside. Not sure if theres something more elegant though

Reply:
Not necessarily. Break statements _can_ work like this; the idea is that it'll just continue to the default case if this IF statement doesn't resolve true. In practice, when using original colours instead of cosmetics ones.

If it's considered bad coding practice then I can rearrange it. Though if-breaks are used elsewhere in the code already.

Reply:
Although now I'm actually questioning myself on that.

Reply:
> Not necessarily. Break statements _can_ work like this; the idea is that it'll just continue to the default case if this IF statement doesn't resolve true. In practice, when using original colours instead of cosmetics ones.
> 
> If it's considered bad coding practice then I can rearrange it. Though if-breaks are used elsewhere in the code already.

Yeah that's what I thought of at first, however, I think it's possible for it to fail the ACTORCAT_NPC condition, but succeed the ACTORCAT_ENEMY condition. This is a result of them using different CVar 'has this cosmetic option changed'

Reply:
Alright, I see what you're saying now. It's the middle of the night here and I'm exhausted, but I'll take a look at it within 24 hours.